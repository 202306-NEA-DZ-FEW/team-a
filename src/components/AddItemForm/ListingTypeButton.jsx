import React from "react";

function ListingTypeButton({ text, selected, onSelect }) {
  return (
    <button
      onClick={() => onSelect(text)}
      type='button'
      className={`${
        selected === text ? "bg-secondary " : "bg-black"
      } px-3 py-2 flex-1 text-sm font-light rounded-lg text-white`}
    >
      {text}
    </button>
  );
}

export default ListingTypeButton;
