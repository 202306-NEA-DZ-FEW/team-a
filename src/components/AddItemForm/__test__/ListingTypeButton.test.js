import renderer from "react-test-renderer";

import ListingTypeButton from "../ListingTypeButton";

it("renders correctly", () => {
  const mockHandleSelect = jest.fn();
  const tree = renderer
    .create(
      <ListingTypeButton text='text' selected='' onSelect={mockHandleSelect} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
