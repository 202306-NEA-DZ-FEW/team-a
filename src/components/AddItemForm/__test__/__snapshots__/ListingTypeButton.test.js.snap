// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="bg-black px-3 py-2 flex-1 text-sm font-light rounded-lg text-white"
  onClick={[Function]}
  type="button"
>
  text
</button>
`;
